includes/wikia/api/SendGridPostBackApiController.class.php

```diff
@@ -1,91 +1,115 @@
 <?php
 /**
- * API which SendGrid will post data to when an email is opened, clicked, bounced back, unsubscribed, marked as spam, etc..
+ * API which SendGrid will post data to when an email is opened, clicked, etc..
  *
- * @author Piotr Molski ( moli@wikia-inc.com )
+ * For more information see:
+ *
+ *   https://sendgrid.com/docs/API_Reference/Webhooks/event.html
  */
 
-class SendGridPostbackController extends WikiaApiController {
+use Wikia\Logger\WikiaLogger;
 
-	public function safe_get($arr, $key, $default = null) {
-		return isset($arr[$key]) ? $arr[$key] : $default;
-	}
+class SendGridPostbackController extends WikiaApiController {
 
+	/**
+	 * Requests from SendGrid are POSTed via this controller.  The POST data contains a JSON
+	 * encoded associative array with the following keys:
+	 *
+	 * - email
+	 * - wikia-email-id
+	 * - wikia-email-city-id
+	 * - wikia-db
+	 * - event
+	 * - url
+	 * - status
+	 * - reason
+	 *
+	 * NOTE: The 'response' key set before returning is for debugging purposes only.  SendGrid
+	 * only requires that we send back a 200 response code.
+	 *
+	 */
 	public function log() {
-		wfProfileIn( __METHOD__ );
-
-		// Sendgrid uses raw JSON POST format, this is how PHP ingests it
-		$data = file_get_contents("php://input");
```

Comment:
Getting and verifying the `$events` array we get from `$data` was moved into its own method, `getEvents`, further below